starter/src/test/java/com/starter/fullstack/rest/InventoryControllerTest.java

```diff
@@ -0,0 +1,72 @@
+package com.starter.fullstack.rest;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.starter.fullstack.api.Inventory;
+import java.util.List;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@RunWith(SpringRunner.class)
+public class InventoryControllerTest {
+
+  @Autowired
+  private MockMvc mockMvc;
+
+  @Autowired
+  private MongoTemplate mongoTemplate;
+
+  @Autowired
+  private ObjectMapper objectMapper;
+
+  private Inventory inventory;
+
+  @Before
+  public void setup() throws Throwable {
+    this.inventory = new Inventory();
+    this.inventory.setId("ID");
+    this.inventory.setName("TEST");
+    // Sets the Mongo ID for us
+    this.inventory = this.mongoTemplate.save(this.inventory);
+  }
+
+  @After
+  public void teardown() {
+    this.mongoTemplate.dropCollection(Inventory.class);
+  }
+
+  /**
+   * Test create endpoint.
+   * @throws Throwable see MockMvc
+   */
+  @Test
+  public void create() throws Throwable {
+    this.inventory = new Inventory();
+    this.inventory.setId("OTHER ID");
+    this.inventory.setName("ALSO TEST");
+    this.mockMvc.perform(post("/inventory")
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .content(this.objectMapper.writeValueAsString(this.inventory)))
+      .andExpect(status().isOk());
+
+    List<Inventory> inv_list = this.mongoTemplate.findAll(Inventory.class);
+    
+    Assert.assertNotEquals("OTHER ID", inv_list.get(1).getId());
```

Comment:
Line 66 and 68 are not really necessary, the DAO test checks the ids, so you don't need to do it here too. Everything else looks perfect, good job!